pkg/coveragedb: TestWriteExtAPICoverageFor failed

@tarasmadan  (for some reason I could not assign the issue to you directly).

https://github.com/google/syzkaller/actions/runs/13304671528/job/37152837632

```
2025-02-13T09:40:27.7427180Z --- FAIL: TestWriteExtAPICoverageFor (0.00s)
2025-02-13T09:40:27.7429354Z     RowIterator.go:59: FAIL:	Stop()
2025-02-13T09:40:27.7505666Z ##[error]FAIL:	Stop()
2025-02-13T09:40:27.7536361Z         		at: [/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/dashboard/app/coverage_test.go:154 /__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/dashboard/app/public_json_api_test.go:326 /__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/dashboard/app/public_json_api_test.go:262]
2025-02-13T09:40:27.7544877Z     RowIterator.go:59: FAIL: 2 out of 3 expectation(s) were met.
2025-02-13T09:40:27.7550324Z ##[error]FAIL: 2 out of 3 expectation(s) were met.
2025-02-13T09:40:27.7554305Z         	The code you are testing needs to make 1 more call(s).
2025-02-13T09:40:27.7563139Z         	at: [/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/coveragedb/mocks/RowIterator.go:59 /syzkaller/.cache/gomod/golang.org/toolchain@v0.0.1-go1.23.6.linux-amd64/src/testing/testing.go:1176 /syzkaller/.cache/gomod/golang.org/toolchain@v0.0.1-go1.23.6.linux-amd64/src/testing/testing.go:1354 /syzkaller/.cache/gomod/golang.org/toolchain@v0.0.1-go1.23.6.linux-amd64/src/testing/testing.go:1684]
2025-02-13T09:40:27.7571839Z FAIL
2025-02-13T09:40:27.7573125Z coverage: 5.8% of statements
```

--- a/pkg/coveragedb/coveragedb.go
+++ b/pkg/coveragedb/coveragedb.go
@@ -416,9 +416,9 @@ func FilesCoverageStream(ctx context.Context, client spannerclient.SpannerClient
 	resCh := make(chan *FileCoverageWithLineInfo)
 	errCh := make(chan error)
 	go func() {
-		defer iter.Stop()
-		defer close(resCh)
 		defer close(errCh)
+		defer close(resCh)
+		defer iter.Stop()
 		if err := readIterToChan(ctx, iter, resCh); err != nil {
 			errCh <- fmt.Errorf("readIterToChan: %w", err)
 		}

